### PR TITLE
Minimize use of autos

### PR DESCRIPTION
**Pull request by swift-kim**
- Use type deduction in a consistent manner.
  - Based on https://google.github.io/styleguide/cppguide.html#Type_deduction.
  - Avoid use of `auto` whenever possible.
  - Some exceptions:
    - The type can clearly be inferred from r-value (e.g. used with `make_unique<>`, `get_if<>`, `reinterpret_cast<>`, `static_pointer_cast<>`).
    - Arguments in a lambda expression.
    - Lambda expression assigned to a local variable.
    - Simple iterator.
    - `std::pair<>` in foreach loop.
- Consistent variable naming.
  - `v` → `element` (in foreach loop)
  - `it` → `iter` (iterators)
- Remove unnecessary `flutter::`.

### CODE REVIEW

File: shell/platform/tizen/tizen_event_loop.cc [hunk 1]
```diff
@@ -59,7 +59,8 @@ void TizenEventLoop::ExecuteTaskEvents() {
 TizenEventLoop::TaskTimePoint TizenEventLoop::TimePointFromFlutterTime(
     uint64_t flutter_target_time_nanos) {
   const auto now = TaskTimePoint::clock::now();
```
Comment (bbrto21) [inline shell/platform/tizen/tizen_event_loop.cc:61]: How about this?

```suggestion
  const TaskTimePoint now = TaskTimePoint::clock::now();
```

Comment (swift-kim) [reply shell/platform/tizen/tizen_event_loop.cc]: Nice catch! I didn't notice the type alias.

Also updated a few other types in the same file.

Comment (bbrto21) [reply shell/platform/tizen/tizen_event_loop.cc]: nice!

**Review by bbrto21 — APPROVED** [review]
I fully agree with the principles that @swift-kim proposed. but I think this principle also needs the approval of other participants of this repository.

File: shell/platform/tizen/flutter_tizen_engine.cc [hunk 2]
```diff
@@ -542,7 +543,7 @@ FlutterRendererConfig FlutterTizenEngine::GetRendererConfig() {
     config.open_gl.gl_external_texture_frame_callback =
         [](void* user_data, int64_t texture_id, size_t width, size_t height,
            FlutterOpenGLTexture* texture) -> bool {
-      auto engine = reinterpret_cast<FlutterTizenEngine*>(user_data);
+      auto* engine = reinterpret_cast<FlutterTizenEngine*>(user_data);
```
Comment (bwikbs) [inline shell/platform/tizen/flutter_tizen_engine.cc:546]: Why did you leave auto  here?

**Review by bwikbs — APPROVED** [review]
I agree with your purpose. :+1:

File: shell/platform/tizen/flutter_tizen_engine.cc (hunk 2 above)
Comment (swift-kim) [reply shell/platform/tizen/flutter_tizen_engine.cc:546]: > - The type can clearly be inferred from r-value

This exception is applied in this case.

**Pull request merged**